lib/utils/network_connection_service.dart

```diff
@@ -0,0 +1,60 @@
+import 'dart:async';
+
+import 'package:connectivity_plus/connectivity_plus.dart';
+import 'package:matrix/matrix.dart';
+
+class NetworkConnectionService {
+  final Connectivity connectivity;
+  late StreamController<ConnectivityResult> networkStreamController;
+  late StreamSubscription<ConnectivityResult> networkStreamSubscription;
+
+  NetworkConnectionService(this.connectivity);
+
+  void onInit() {
+    _setupNetworkStream();
+    _listenNetworkConnectionChanged();
+  }
+
+  void onDispose() {
+    networkStreamController.close();
+    networkStreamSubscription.cancel();
+  }
+
+  void _setupNetworkStream() {
+    networkStreamController = StreamController<ConnectivityResult>.broadcast();

```

Comment:
Why we need it? Why not directly listen from connectivity?

Reply:
Because onConnectivityChanged stream only emits when there is a change. When we change view, the stream needs to have the current status. So IMO it's better to have one stream that is synchronized then copied in every streambuilder (which manage his own lifeycle) we need.

We could used checkConnectivity() to initialize and merge with onChanged but then we would have needed FutureBuilder in a streambuilder in every widget which is not convenient.

I'm open to any suggestions